vital/util/span.h

```diff
@@ -0,0 +1,153 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/// \file
+/// \brief Definition of span template class.
+
+#include <vital/util/vital_util_export.h>
+
+#include <algorithm>
+#include <stdexcept>
+#include <type_traits>
+
+#include <cmath>
+
+namespace kwiver {
+
+namespace vital {
+
+// ----------------------------------------------------------------------------
+/// Represents a numerical span with an upper and lower bound.
+template < class T >
+class VITAL_UTIL_EXPORT span
```

Comment:
So... I do not, unfortunately, have any recommendations offhand, but I wonder if we might want to name this something else, to avoid confusion with [`std::span`](https://en.cppreference.com/w/cpp/container/span), which represents something rather different (a subset of `vital::range::range`, in fact).

Reply:
`numeric_span`, perhaps? It's a bit wordier...

Reply:
As mentioned over other channels, Boost already has [`boost::interval`](https://www.boost.org/doc/libs/1_77_0/libs/numeric/interval/doc/interval.htm), which appears to serve substantially the same purpose. Between that precedence (and corresponding reduced risk that the standard library will later adopt the same name to mean something totally different) and given there is also an argument to be made for that name being the "correct" mathematical term, I think there is a strong argument for us to (also) use "interval".